### ISSUE TITLE
Unexpected `java.lang.ClassCastException` when decoding `immutable.ArraySeq`

### ISSUE THREAD

**Issue by plokhotnyuk** [FasterXML/jackson-module-scala#588]
Steps to reproduce:
```
git clone --depth 1 https://github.com/plokhotnyuk/jsoniter-scala
cd jsoniter-scala
git checkout github.com/FasterXML/jackson-module-scala/issues/588
sbt jsoniter-scala-benchmarkJVM/test
```

**Comment by pjfanning** [comment]
@plokhotnyuk thanks for reporting this. I've started #589 and so far the issue is not obvious but appears to affect immutable ArraySeqs but not not mutable ones. I will continue to investigate later today.

**Comment by plokhotnyuk** [comment]
@pjfanning Thanks for the fast response!

Could it be due to missing of `immutable.ArraySeq` in factory lists here? https://github.com/FasterXML/jackson-module-scala/blob/2.14/src/main/scala-2.13/com/fasterxml/jackson/module/scala/deser/SeqDeserializerModule.scala

**Comment by pjfanning** [comment]
@plokhotnyuk yes, that looks to be the case - I had just found it when I saw your comment - I should be able to merge the fix soon (barring unforeseen CI failures)

**Comment by pjfanning** [comment]
@plokhotnyuk the 2.14.0-SNAPSHOT with the fix is on Sonatype - I've also backported the fix to 2.13.3+7-9efbe43c-SNAPSHOT.

**Comment by pjfanning** [comment]
Fix is in the v2.13.4 release
